<?xml version="1.0" ?>
<launch>

  <arg name="serial_no"		default=""/>
  <arg name="camera_name"	default="a_bot_camera"/>
  <arg name="camera_width"	default="640"/>
  <arg name="camera_height"	default="480"/>
  <arg name="fps"		default="30"/>
  <arg name="enable_streaming"	default="true"/>
  <arg name="manager"		default=""/>
  <arg name="publish_tf"	default="true"/>
  <arg name="vis"		default="true"/>

  <include file="$(find realsense2_camera)/launch/rs_camera.launch">
    <arg name="serial_no"		value="$(arg serial_no)"/>
    <arg name="camera"			value="$(arg camera_name)"/>
    <arg name="tf_prefix"		value="calibrated_$(arg camera_name)"/>
    <arg name="publish_tf"		value="$(arg publish_tf)"/>
    <arg name="enable_depth"		value="true"/>
    <arg name="enable_infra1"		value="true"/>
    <arg name="enable_infra2"		value="false"/>
    <arg name="enable_pointcloud"	value="false"/>
    <arg name="align_depth"		value="true"/>
    <arg name="depth_width"		value="$(arg camera_width)"/>
    <arg name="depth_height"		value="$(arg camera_height)"/>
    <arg name="depth_fps"		value="$(arg fps)"/>
    <arg name="infra_width"		value="$(arg camera_width)"/>
    <arg name="infra_height"		value="$(arg camera_height)"/>
    <arg name="infra_fps"		value="$(arg fps)"/>
    <arg name="color_width"		value="$(arg camera_width)"/>
    <arg name="color_height"		value="$(arg camera_height)"/>
    <arg name="color_fps"		value="$(arg fps)"/>
    <arg name="enable_streaming"	value="$(arg enable_streaming)"/>
    <arg name="manager"			value="$(arg manager)"
	 unless="$(eval manager == '')"/>
  </include>

  <include file="$(find aist_depth_filter)/launch/run.launch">
    <arg name="camera_info_topic" value="$(arg camera_name)/color/camera_info"/>
    <arg name="image_topic"	  value="$(arg camera_name)/color/image_raw"/>
    <arg name="depth_topic"	  value="$(arg camera_name)/aligned_depth_to_color/image_raw"/>
    <arg name="normal_topic"	  value=""/>
    <arg name="manager"		  value="$(arg manager)"/>
  </include>

  <node name="set_window_radius" pkg="dynamic_reconfigure" type="dynparam"
	args="set depth_filter window_radius 2"/>

  <group if="$(arg vis)">
    <node name="$(anon rviz)" pkg="rviz" type="rviz" output="screen"
	  args="-d $(find aist_depth_filter)/launch/realsense.rviz"/>

    <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure"/>
  </group>

</launch>
